Source/Orts.Simulation/Simulation/Signalling/CsSignalScript.cs

```diff
@@ -0,0 +1,170 @@
+﻿using Orts.Formats.Msts;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orts.Simulation.Signalling
+{
+    // The exchange of information is done through the TextSignalAspect property.
+    // The MSTS signal aspect is only used for TCS scripts that do not support TextSignalAspect.
+    public abstract class CsSignalScript
+    {
+        // References and shortcuts. Must be private to not expose them through the API
+        private SignalHead SignalHead { get; set; }
+        private SignalObject SignalObject => SignalHead.mainSignal;
+        private int SigFnIndex(string sigFn) => SignalObject.signalRef.ORTSSignalTypes.IndexOf(sigFn);
+        private SignalObject SignalObjectById(int id) => SignalObject.signalRef.SignalObjects[id];
+
+        // Public interface
+        public string DebugFileName = String.Empty;
```

Comment:
Let's document the public interface with docstrings (`/// <summary> ... </summary>`).

Reply:
Sure! I have half of them already documented in my working copy, as well as the interface with TCS